src/plugins/fs/generic.c

```diff
@@ -474,6 +503,8 @@ static gboolean device_operation (const gchar *device, BDFsOpType op, guint64 ne
                 return bd_fs_ext4_check (device, NULL, error);
             case BD_FS_LABEL:
                 return bd_fs_ext4_set_label (device, label, error);
+            case BD_FS_SIZE:
+                g_assert_not_reached ();
```

Comment:
I'm usually strongly against asserts in libraries but this is not a user input validation. If something ends here, it's a programmer error (and we need to cover all cases in the switch to make gcc happy). Thoughts?

Reply:
I'd add a single assert to the beginning of the function to assert that `op != BD_FS_GET_SIZE`. And if GCC is not clever enough to take that as covered for the `switch` then just suppress the warning locally.

Reply:
It's not only gcc. Also coverity and cppcheck and I don't want to add ten lines of comments and pragmas :)

Reply:
Abort is just like a segfault - let the process fail, grab a backtrace and let developers fix it. That's in case of a programmer error, user input should be handled somewhere else.

Anyway, you may want to validate input at the very beginning of the function via `g_warn_if_fail()`/`g_return_val_if_fail()` and probably also use `g_warn_if_reached()` instead of a hard assert.

Reply:
> Abort is just like a segfault - let the process fail, grab a backtrace and let developers fix it.

Isn't that exactly what we want here? This is not user input. It's an invalid call of a `static` function.